app/src/behaviors/behavior_toggle_layer.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020 Cody McGinnis <brainwart@gmail.com>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_toggle_layer
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct behavior_tog_config { };
+struct behavior_tog_data { };
+
+static int behavior_tog_init(struct device *dev)
+{
+	return 0;
+};
+
+
+static int tog_keymap_binding_pressed(struct device *dev, u32_t position, u32_t layer, u32_t _)
+{
+  LOG_DBG("position %d layer %d current %d", position, layer, zmk_keymap_layer_active(layer));
+
+  if (zmk_keymap_layer_active(layer))
+    return zmk_keymap_layer_deactivate(layer);
+
+  return zmk_keymap_layer_activate(layer);
+}
+
+static int tog_keymap_binding_released(struct device *dev, u32_t position, u32_t layer, u32_t _)
```

Comment:
This is necessary for now, but I think we likely need to change it so behaviors aren't required to provide both callbacks, and the keymap logic will do the right now.